portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -85,7 +84,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     String userName = request.getParameter("user");
     String comment = request.getParameter("comment-body");
     // it is allowable for the image url to be null
-    String blobKey = getBlobKey(request, "image").getKeyString();
+    BlobKey currentKey = getBlobKey(request, "image");
+    String blobKey;
+    if (currentKey == null) {
+      blobKey = null;
+    } else {
```

Comment:
You can initialize `blobKey` to null, then just have a single `if` block for the non-null case.